### PR TITLE
Feature/430 profile pictures

### PR DESCRIPTION
**Pull request by Icaro212**

### CODE REVIEW

File: bugalink-frontend/pages/index.tsx [hunk 1]
```diff
@@ -43,7 +45,7 @@ export default function Home() {
             className="aspect-square h-14"
             href={NEXT_ROUTES.PROFILE(user?.user_id)}
           >
-            {user ? <Avatar src={user.photo} /> : <AvatarSkeleton />}
+            {user ? <Avatar src={userPhoto?.photo} /> : <AvatarSkeleton />}
```
Comment (andrsdt) [inline bugalink-frontend/pages/index.tsx:48]: I believe you are doing this to solve the problem of an outdated JWT token when updating the profile picture, so you call the API every time to bring the user's current photo

File: bugalink-frontend/pages/users/[id]/edit.tsx [hunk 2]
```diff
@@ -85,9 +92,26 @@ export default function EditProfile({ data }) {
...
+        const formData = new FormData();
+        formData.append('photo', file);
+        formData.append('first_name', name);
+        formData.append('last_name', surname);
+
+        
+        axiosAuth
+          .put(url, formData)
+          .then((res) => {
+            signOut({
```
Comment (andrsdt) [inline bugalink-frontend/pages/users/[id]/edit.tsx:106]: But you are also logging out the user everytime they edit their profile (and potentially their photo). Then, you don't need to do the call to `useUser()` in the home page, that's an unnecessary call to the backend. Ideally, remove the `userPhoto` variable in the home page and keep it as it was

File: bugalink-frontend/pages/users/[id]/edit.tsx [hunk 3]
```diff
@@ -85,9 +92,26 @@ export default function EditProfile({ data }) {
...
+
+        const formData = new FormData();
+        formData.append('photo', file);
+        formData.append('first_name', name);
+        formData.append('last_name', surname);
+
+        
+        axiosAuth
+          .put(url, formData)
+          .then((res) => {
```
Comment (andrsdt) [inline bugalink-frontend/pages/users/[id]/edit.tsx:105]: using async/await is often cleaner and prefered over `.then()` and `.catch()` as that's our code standard

File: bugalink-backend/users/models.py [hunk 4]
```diff
@@ -1,6 +1,20 @@
...
+    # print(instance.user)
+    return "users/{0}/{1}".format(instance.id, filename)
+
+class OverwriteStorage(FileSystemStorage):
+    def get_available_name(self, name, max_length=None):
+        original_name=self.url(name).split(".")[0]
+        new_fileName = name.split(".")[0]
+        if original_name.__contains__(new_fileName):
+            self.delete(name)
+            print("Estoy aqui")
```
Comment (andrsdt) [inline bugalink-backend/users/models.py:15]: You forgot to remove the debugging prints

File: bugalink-backend/users/models.py [hunk 5]
```diff
@@ -1,6 +1,20 @@
...
 from django.db import models
 
+def user_directory_path(instance, filename):
+    # print(instance.user)
+    return "users/{0}/{1}".format(instance.id, filename)
+
+class OverwriteStorage(FileSystemStorage):
+    def get_available_name(self, name, max_length=None):
+        original_name=self.url(name).split(".")[0]
+        new_fileName = name.split(".")[0]
```
Comment (andrsdt) [inline bugalink-backend/users/models.py:12]: you are mixing snake_case and camelCase in the same variable, please follow the conventions

File: bugalink-backend/users/models.py [hunk 6]
```diff
@@ -1,6 +1,20 @@
 from django.contrib.auth.models import AbstractBaseUser, BaseUserManager
+from django.core.files.storage import FileSystemStorage
 from django.db import models
 
+def user_directory_path(instance, filename):
+    # print(instance.user)
+    return "users/{0}/{1}".format(instance.id, filename)
+
+class OverwriteStorage(FileSystemStorage):
```
Comment (andrsdt) [inline bugalink-backend/users/models.py:9]: You have fixed the problem of redundant files (avatar.png, avatar.jpeg) in `users/models.py` but it's still occuring in the `class OverwriteStorage()` in `driver/models.py`. Fix it there too if possible

File: bugalink-backend/users/models.py [hunk 7]
```diff
@@ -1,6 +1,20 @@
 from django.contrib.auth.models import AbstractBaseUser, BaseUserManager
+from django.core.files.storage import FileSystemStorage
 from django.db import models
 
+def user_directory_path(instance, filename):
+    # print(instance.user)
```
Comment (andrsdt) [inline bugalink-backend/users/models.py:6]: Remove commented print code. You should have been noticed that bad smell by Sonarlint if you have it installed in your code editor

File: bugalink-backend/users/views.py [hunk 8]
```diff
@@ -74,7 +74,7 @@ def delete(self, request, *args, **kwargs):
         user.email = new_email
         user.first_name = "Usuario eliminado"
         user.last_name = ""
-        user.photo = ""
+        user.photo = user.photo.delete()
```
Comment (andrsdt) [inline bugalink-backend/users/views.py:77]: You are deleting the file and assigning it to a variable? Remove the `user.photo = ` part and only keep the code for deleting

File: bugalink-frontend/next.config.js [hunk 9]
```diff
@@ -12,4 +12,8 @@ module.exports = withPWA({
 
     return config;
   },
+  images: {
+    domains: ['localhost', '127.0.0.1', 'bugalink.es'],
```
Comment (andrsdt) [inline bugalink-frontend/next.config.js:16]: The domain is actually `app.bugalink.es`, this might not work in the deployment. Keep an eye on it

**Review by andrsdt — CHANGES_REQUESTED** [review]
The functionality seems to work but there are many bad smells and parts of the code that aren't following the conventions. Fix them before merging to avoid making the spaghetti code snowball bigger 😅

File: bugalink-frontend/pages/users/[id]/edit.tsx (hunk 2 above)
Comment (Icaro212) [reply bugalink-frontend/pages/users/[id]/edit.tsx:106]: Again reffer to my previous comment

File: bugalink-backend/users/models.py (hunk 6 above)
Comment (Icaro212) [reply bugalink-backend/users/models.py:9]: Will do

File: bugalink-frontend/pages/index.tsx (hunk 1 above)
Comment (andrsdt) [reply bugalink-frontend/pages/index.tsx:48]: Makes sense. However, we have the backend URL available as an environment variable (`process.env.NEXT_PUBLIC_BACKEND_URL`) so we can use it to concatenate it with the given `photo` variable and avoid making the unnecessary API call

**Comment by Icaro212** [comment]
Run npm i before trying out dotenv is necesary for the frontend now

**Pull request merged**